File: golang/2490.circular-sentence.go
import "strings"

func isCircularSentence(sentence string) bool {
	words := strings.Split(sentence, " ")
	n := len(words)

	for i := 0; i < n; i++ {
		if words[i][len(words[i])-1] != words[(i+1)%n][0] {
			return false
		}
	}

	return true
}

File: cpp/2490.circular-sentence.cpp
#include <string>
using std::string;

class Solution {
public:
  bool isCircularSentence(string sentence) {

    if (sentence.front() != sentence.back()) {
      return false;
    }
    int i = 0, n = sentence.size();

    while (i < n) {
      if (sentence[i] != ' ') {
        i++;
        continue;
      }

      // if (i == n - 1 || i == 0) {
      //   return false;
      // }

      if (sentence[i - 1] != sentence[i + 1]) {
        return false;
      }
      i++;
    }

    return i == n;
  }
};